### ISSUE TITLE
Feature request: Renaming enum abstract values

### ISSUE THREAD

**Issue by markknol** [fdorg/flashdevelop#1465]
In Haxe you can define a [`enum` type](https://haxe.org/manual/types-enum-instance.html) but there is also a [`@:enum abstract`](https://haxe.org/manual/types-abstract-enum.html) which can be used to define finite value sets: 

```haxe
@:enum abstract MyEnum(String) {
  var VALUE1 = "value1";
  var VALUE2 = "value2";
  var VALUE3 = "value3";
}
```
This is a nice, typed alternative to this:
```haxe
class PoorManEnum {
  public static inline var VALUE1 = "value1";
  public static inline var VALUE2 = "value2";
  public static inline var VALUE3 = "value3";
}
```

HaxeDevelop gives the option to rename the values of the `@:enum abstract`, but it basically never does anything with it in the classes that use it. I think you can assume the fields in this enum abstract are static, I think it now treads it as a class member, because it only renames itself.

It would be great if FlashDevelop/HaxeDevelop supports renaming enum abstracts. Thanks!

**Comment by SlavaRa** [comment]
I think it's possible to do it, but I need time to write the tests

**Comment by SlavaRa** [comment]
depends on #1772

**Comment by SlavaRa** [comment]
fixed #2143
